### DISCUSSION TITLE
How to pass Object arguments in constructor when using forge create

### DISCUSSION THREAD

**Discussion by 0xautohub** [pavlovcik/uad-ui#360] [Q&A]
Simple deployment command of a contract with two constructor arguments using `forge` is as follows<br/>
`forge create src/Contract.sol:MyToken --constructor-args "My Token" "MT"`

But need to pass some objects as constructor arguments to deploy [a contract](https://github.com/ubiquity/ubiquity-dollar/pull/313/commits/5f44b39ad501fe91a3284add5bbe9555cefee198#diff-c872256a2a42ae8cc810e20f66c68b41739206007cd17c6a80845c8af7eaaba7R20) similar to below one

```
struct DiamondArgs {
    address owner;
    address init;
    bytes initCalldata;
}

contract MyContract {
    constructor(DiamondArgs memory _args) {}
}
```

So, the deployment command would look like [this](https://github.com/ubiquity/ubiquity-dollar/pull/313/commits/cecc893f44bdb57649845db615cc163933e523f5#diff-3721e44e713d393b90572e9040b5b18d95aae3cf6b0b8e41588441995cd2e684R44)<br/>
`forge create src/MyContract.sol:MyContract --constructor-args {owner: "ownerAddress", init: "initAddress", initCalldata: "initCalldata"}` <br/>

When trying to pass that object, the result looks like this :<br/>
`forge create --constructor-args [object Object] ...`<br/>

What's the workaround for this?

**Comment by 0x4007** [comment]
This sort of question should be posted in Q&A under discussions. I just transferred!

**Comment by 0x4007** [comment]
ChatGPT says:

To pass objects as constructor arguments to a contract using forge, you can encode the object as JSON and pass the JSON string as an argument to the --constructor-args flag. For example, in the case of the DiamondArgs struct from your example, the deployment command would look something like this:

```
forge create src/MyContract.sol:MyContract --constructor-args '{"owner": "ownerAddress", "init": "initAddress", "initCalldata": "initCalldata"}'
```
Alternatively, you can also use the --json flag to pass the object as a JSON file. The command would look something like this:

```
forge create src/MyContract.sol:MyContract --json diamond-args.json
```
where diamond-args.json is the name of a file that contains the JSON-encoded DiamondArgs object.

In either case, the DiamondArgs object will be passed to the MyContract contract's constructor as a single argument when the contract is deployed.

**Reply by 0x4007** [reply]
It looks like you didn't stringify your JSON object

**Reply by 0xautohub** [reply]
> forge create src/MyContract.sol:MyContract --constructor-args '{"owner": "ownerAddress", "init": "initAddress", "initCalldata": "initCalldata"}'

Tried this with simple contract with object as constructor arguments, but no hope

![image](https://user-images.githubusercontent.com/5579669/205669301-d894d984-d59a-463d-af85-7e96d67dfeae.png)


![image](https://user-images.githubusercontent.com/5579669/205666798-c14822e5-dff5-4bdb-a26d-db07ae5afba1.png)
